src/iptc.cpp

```diff
@@ -110,6 +195,15 @@ namespace Exiv2 {
 
     std::ostream& Iptcdatum::write(std::ostream& os, const ExifData*) const
     {
+        const StringValue* val = dynamic_cast<const StringValue*>(&value());
+        if (val != NULL) {
+            // In case of being a StringValue, it could have non UTF-8 which we want to supress
+            if (is_utf8(val->toString())) {
+                return os << value();
+            } else {
+                return os << N_("(Non UTF-8 bytes supressed)");
```

Comment:
How about printing the bytes as a hex-string? That way you at least get something.

Reply:
I think the @clanmills 's opinion will worth than mine. I do not know if it makes sense to print those value in hexadecimal, or we could have big arrays of non UTF-8 chars in some cases.